src/main/java/frc/robot/subsystems/Dashboard.java

```diff
@@ -0,0 +1,376 @@
+    private final GenericEntry _flywheelVelocityRange;
+    private final GenericEntry _maxFlywheelSpeed;
+    private final GenericEntry _flywheelIntakeSpeed;
+
+    /*
+     * Subsystems
+     */
+    private final ShooterBed      _shooterBed;
+    private final Notepath        _notepath;
+    private final Intake          _intake;
+    private final ShooterFlywheel _shooterFlywheel;
+    private final Drive           _drive;
+    private final Climb           _climb;
+
+    /*
+     * SendableChoosers for Autonomous options
+     */
+    private final SendableChooser<Integer> _autoDelayChooser;
+
+    public Dashboard(ShooterBed shooterBed, Notepath notepath, ShooterFlywheel shooterFlywheel, Drive drive, Intake intake, Climb climb)
+    {
+        _drive           = drive;
+        _shooterBed      = shooterBed;
+        _notepath        = notepath;
+        _shooterFlywheel = shooterFlywheel;
+        _intake          = intake;
+        _climb           = climb;
+
+        /*
+         * DASHBOARD
+         */
+        var dashboard = Shuffleboard.getTab("Dashboard");
+
+        // Camera Stream
+        dashboard.addCamera("Camera Stream", Constants.Vision.CAMERA_NAME, Constants.Vision.CAMERA_URL).withWidget(BuiltInWidgets.kCameraStream).withPosition(0, 0).withSize(15, 10)
+                .withProperties(Map.of("Show crosshair", false, "Show controls", false));
+
+        // Field
+        _field = new Field2d();
+        dashboard.add("Field", _field).withPosition(15, 0).withSize(15, 8);
+
+        // Color box
+        var colorBoxLayout = dashboard.getLayout("Color Box", BuiltInLayouts.kGrid).withPosition(15, 12).withSize(5, 3).withProperties(Map.of("Number of columns", 1, "Number of rows", 2, "Label position", "LEFT"));
+        _allianceBox = colorBoxLayout.add("Alliance", false).withPosition(0, 0).withWidget(BuiltInWidgets.kBooleanBox).withProperties(Map.of("Color when true", "blue", "Color when false", "red")).getEntry();
+        _hasNote     = colorBoxLayout.add("Has Note", false).withPosition(0, 1).withWidget(BuiltInWidgets.kBooleanBox).withProperties(Map.of("Color when true", "green", "Color when false", "red")).getEntry();
+
+        // Swerve Module Angles
+        var driveSettingsLayout = dashboard.getLayout("Swerve Angles", BuiltInLayouts.kGrid).withPosition(15, 8).withSize(5, 4).withProperties(Map.of("Number of columns", 2, "Number of rows", 2));
+        _flAngle = driveSettingsLayout.add("FL Angle", 0.0).withPosition(0, 0).withSize(2, 2).withWidget(BuiltInWidgets.kTextView).getEntry();
+        _frAngle = driveSettingsLayout.add("FR Angle", 0.0).withPosition(1, 0).withSize(2, 2).withWidget(BuiltInWidgets.kTextView).getEntry();
+        _blAngle = driveSettingsLayout.add("BL Angle", 0.0).withPosition(0, 1).withSize(2, 2).withWidget(BuiltInWidgets.kTextView).getEntry();
+        _brAngle = driveSettingsLayout.add("BR Angle", 0.0).withPosition(1, 1).withSize(2, 2).withWidget(BuiltInWidgets.kTextView).getEntry();
```

Comment:
We don't need `.withSize` on any of these angles:

```suggestion
        _flAngle = driveSettingsLayout.add("FL Angle", 0.0).withPosition(0, 0).withWidget(BuiltInWidgets.kTextView).getEntry();
        _frAngle = driveSettingsLayout.add("FR Angle", 0.0).withPosition(1, 0).withWidget(BuiltInWidgets.kTextView).getEntry();
        _blAngle = driveSettingsLayout.add("BL Angle", 0.0).withPosition(0, 1).withWidget(BuiltInWidgets.kTextView).getEntry();
        _brAngle = driveSettingsLayout.add("BR Angle", 0.0).withPosition(1, 1).withWidget(BuiltInWidgets.kTextView).getEntry();
```